### ISSUE TITLE
type[T] incompatible with type | Any

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#3118]
**Describe the bug**
```python
from __future__ import annotations

from typing import Any, TypeVar

T = TypeVar("T")


def f(x: type[T]) -> T:
    ...


def g() -> type | Any:
    ...


y = g()

# type error Argument of type "type | Any" cannot be assigned to parameter "x" of type "Type[T@f]" in function "f"
# Type "type | Any" cannot be assigned to type "Type[T@f]"
f(y)
```

**Expected behavior**
No type errors. Both type and Any are compatible with type[T].

**VS Code extension or command-line**
pyright cli. 1.1.225 shows error and doing a bisect it comes from 1.1.158. Versions before that show no error.

**Comment by hmc-cs-mdrissi** [comment]
I encountered a second similar version of this error. Not sure if it's same issue or slightly different one. A union of types isn't matching a type[T] this time without Any involved.

```python
from __future__ import annotations

from typing import TypeVar

T = TypeVar('T')

def identity(x: type[T]) -> type[T]:
    return x

def f(cls: type) -> None:
    if issubclass(cls, int):
        ...
    
   # False positive error: 
   # Argument of type "Type[int] | type" cannot be assigned 
   # to parameter "x" of type "Type[T@identity]" in function "identity" 
    identity(cls)
```

**Comment by erictraut** [comment]
This will be fixed in the next release.
